### ISSUE TITLE
2 questions

### ISSUE THREAD

**Issue by zygym** [icesat-2UT/PhoREAL#1]
1.
![image](https://user-images.githubusercontent.com/62503050/86106314-00b67c00-baf3-11ea-8726-a2ecdc36d1a4.png)
I am a Windows user, and the software are shown in my computer as above, and I have no idea about the function of the "get reference data input" and "find icesat-2 offsets relative to reference data", if you can offer me some help, that will be greatful! Thank you!
2.
![image](https://user-images.githubusercontent.com/62503050/86106876-b1bd1680-baf3-11ea-863d-83be290270fa.png)
![image](https://user-images.githubusercontent.com/62503050/86106919-c3062300-baf3-11ea-9b74-3959e8b9a182.png)
I operated my data as the "PhoREAL_User_Manual.pdf" guide, but the result of mine just different from the examples you provided, besides, the method to generate dsm.tif and dtm.tif are also a problem for me,  if you can offer me some help, that will be greatful! Thank you!

**Comment by icesat-2UT** [comment]
Hi,

Thank you for using PhoREAL to help process your data. Here are responses to your questions:


1)     The version you’ve downloaded is newer than the documentation, so the user interface looks different. But most functionality is the same, just with some new features. The “Get Reference Data Input” section is meant for use whenever you (the user) have your own “truth” (or “reference”) point cloud data. The ICESat-2 data is the “measured” data. One of the things that we tried to make available for users is a method to correct ICESat-2 data with respect to some provided “reference” data or vice versa. We have access to some reference data at our facility, but we are not allowed to release that data publicly. So any reference data for this section must be provided by you, the user. The “Find Offsets” section will actually run through a quick algorithm that will align the “measured” ICESat-2 data to your “reference” data. If you do not intend to use these sections, you can just uncheck them in the upper-right hand checkbox in each section. At the time of this writing, there are some issues with loading in multiple “reference” point clouds. This is not an issue when we run PhoREAL internally, but it is for external users as it requires some extra data files that are not public yet. So for the time being, that full capability is not available yet. However, you can load in 1 reference .tif file.

2)     For you second question, I am having some trouble understanding what is wrong with your data or what particular issue you are having. Can you elaborate? If you load in ICESat-2 data, you should be able to plot it within the PhoREAL GUI or analyze the output data in Google Earth (.kml file), any csv viewer like Microsoft Excel (.csv file), or any point cloud viewer like Cloud Compare or QT Modeler (.las file). The ICESat-2 ATL03 data will appear as a line segment (transect) since that is how the data is collected by ICESat-2.

Thanks again,
Mike Alonzo

**Comment by zygym** [comment]
Thank you for your responses! I understood question 1, thanks! What I want to explain in question 2 is that the information of the .las file you provided as an example are more specific than I generate by PhoREAL. I can see trees and other features in your .las file, and there are no such specific information in my .las file. Thank you again!






------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"icesat-2UT"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月30日(星期二) 晚上9:20
收件人:&nbsp;"icesat-2UT/PhoREAL"<PhoREAL@noreply.github.com&gt;;
抄送:&nbsp;"Born To Beyond"<2449477623@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [icesat-2UT/PhoREAL] 2 questions (#1)





 Hi, 
 
 Thank you for using PhoREAL to help process your data. Here are responses to your questions: 
 
 
 1)     The version you’ve downloaded is newer than the documentation, so the user interface looks different. But most functionality is the same, just with some new features. The “Get Reference Data Input” section is meant for use whenever you (the user) have your own “truth” (or “reference”) point cloud data. The ICESat-2 data is the “measured” data. One of the things that we tried to make available for users is a method to correct ICESat-2 data with respect to some provided “reference” data or vice versa. We have access to some reference data at our facility, but we are not allowed to release that data publicly. So any reference data for this section must be provided by you, the user. The “Find Offsets” section will actually run through a quick algorithm that will align the “measured” ICESat-2 data to your “reference” data. If you do not intend to use these sections, you can just uncheck them in the upper-right hand checkbox in each section. At the time of this writing, there are some issues with loading in multiple “reference” point clouds. This is not an issue when we run PhoREAL internally, but it is for external users as it requires some extra data files that are not public yet. So for the time being, that full capability is not available yet. However, you can load in 1 reference .tif file. 
 
 2)     For you second question, I am having some trouble understanding what is wrong with your data or what particular issue you are having. Can you elaborate? If you load in ICESat-2 data, you should be able to plot it within the PhoREAL GUI or analyze the output data in Google Earth (.kml file), any csv viewer like Microsoft Excel (.csv file), or any point cloud viewer like Cloud Compare or QT Modeler (.las file). The ICESat-2 ATL03 data will appear as a line segment (transect) since that is how the data is collected by ICESat-2. 
 
 Thanks again, 
 Mike Alonzo

**Comment by zygym** [comment]
Thank you for your responses! I understood question 1, thanks! What I want to explain in question 2 is that the information of the .las file you provided as an example are more specific than I generate by PhoREAL. I can see trees and other features in your .las file, and there are no such specific information in my .las file. Thank you again!
![image](https://user-images.githubusercontent.com/62503050/86248717-d4792900-bbe0-11ea-9630-e6ce28b799e5.png)
![image](https://user-images.githubusercontent.com/62503050/86248809-f672ab80-bbe0-11ea-94b0-f351dbdfbb13.png)

**Comment by icesat-2UT** [comment]
Hi,

I understand your question, thank you for your response.

If you use only an ATL03 file, there will be no classified photons in your output .las file. Only signal confidence is provided in the .las file (which comes from ATL03). But if you provide a corresponding ATL08 file along with your ATL03 file, then there will be classifications for ground, low vegetation, and high vegetation. This information comes from ATL08 and gets mapped back to the ATL03 point cloud file.

Let me know if this helps you.

Thanks,
Mike Alonzo

**Comment by zygym** [comment]
Thank you. Actually, I have the ATL08 file correspond to ATL03 file, the information I operate  the PhoREAL and the results generate by PhoREAL are shown as below. How to combine the information of ATL08 and ATL03 together? Thank you very much!
![image](https://user-images.githubusercontent.com/62503050/86250221-e5c33500-bbe2-11ea-9461-ba029ba71c51.png)

![image](https://user-images.githubusercontent.com/62503050/86250166-d348fb80-bbe2-11ea-8966-6a102e5aa434.png)

**Comment by icesat-2UT** [comment]
I see. The classifications from the ATL08 file get mapped back to each ATL03 photon and that information is embedded into your output .las file. They should be stored in the ‘classifications’ field of the point cloud file. Are you able to see that?

**Comment by zygym** [comment]
Yes! I found some useful information from the "classification", but I still have a question about why the data of mine is so "slim", just like a line, not so wide like the example you provided. Thank you.
![image](https://user-images.githubusercontent.com/62503050/86257452-23788b80-bbec-11ea-9c35-70f2fbe8eb80.png)

**Comment by icesat-2UT** [comment]
The data I provided is showing “truth” (or “reference”) data along with the ICESat-2 data. The ICESat-2 data is just a line as it is a laser pulse directed completed nadir from the ICESat-2 satellite while in orbit. The satellite does not scan in the cross-track direction, it only measures in the along-track direction that the satellite is moving. We typically calls this data “transects” as they are a 2-dimensional line, moving in the along-track direction.

This transect data from ICESat-2 should overlay nicely on a “truth” or “reference” point cloud of the same area that you have to provide.

Hope this helps!

Thanks,
Mike Alonzo

**Comment by zygym** [comment]
Thanks! That is to say if I want to generate a .las file like the example you provided I need some extra point cloud data(like airborne point cloud data) overlay with ATL03, is that right? Besides, the .tif files you provided are "reference" files or generated by the PhoREAL? Thank you very much!
![image](https://user-images.githubusercontent.com/62503050/86353973-c4be1b00-bc9a-11ea-8d24-98f64516511f.png)

**Comment by icesat-2UT** [comment]
Yes, unfortunately you will have to provide any extra reference point cloud data where you can overlay ATL03 data on top of. Currently, our reference data sets are not public. The same goes for any reference .tif file. Any reference data that we show in examples is not from PhoREAL, only the ATL03 transect data is from PhoREAL. All other .las/.tif files shown are examples of how ICESat-2 ATL03 data looks when laid over reference data.

If you find a reference data set and overlay the ATL03 data on top of it and notice that it does not line up very closely, then you can load the reference data into PhoREAL along with the ATL03 data and it will attempt to geolocate the ATL03 data with respect to the reference data for a closer fit.

Thanks,
Mike Alonzo

**Comment by zygym** [comment]
![image](https://user-images.githubusercontent.com/62503050/86425229-2754fc80-bd17-11ea-8e7c-a03d15200fb0.png)
I see, thank you very much! When I check the .csv file I met some new problems. What is "-1" represent in  "Classification"? The "Height" colum represents the absolute height(including altitute) or the height relative to the ground?

**Comment by icesat-2UT** [comment]
For Classifications:
-1 = Unclassified ATL03 point
1 = ATL03 Ground point
2 = ATL03 Canopy point
3 = ATL03 Top of Canopy point

Height is the height of the ATL03 photon with respect to WGS-84 Height Above Ellipsoid (HAE). This is also known as ellipsoidal height. This data comes straight from the ATL03 data product and is not altered.

**Comment by zygym** [comment]
Hello, I have a question not relative to the PhoReal but about why the ICESat-2 have two diffierent kinds of beams(strong and weak)? For save energy or other reasons? I brosed the https://nsidc.org/data/icesat-2, and did not found answer. If you can provide me some information, that will be greatful! Thank you.

**Comment by icesat-2UT** [comment]
You can reference this paper:

https://www.sciencedirect.com/science/article/pii/S003442571930344X?via%3Dihub

Written by the director in charge of ICESat-2 at NASA. Specifically, you may be referring to this part of the paper in Section 2.1.1:

Approximately 80% of the laser pulse energy is partitioned into the six primary outgoing laser beams, while 20% is lost to higher-order modes. At the nominal laser power setting, this means that ~660 microjoules (μJ) of the ~835 μJ pulse is used, and ~175 μJ are lost to higher-order modes. The total available laser energy precluded the scenario of having six strong beams; as a result the six primary beams generated by the DOE have unequal energy, with three relatively strong beams and three relatively weak beams. Given the energy losses along the optical path in the laser transmission, the strong beams each contain ~21% of the transmitted energy (~175 μJ per pulse) and the weak beams share the remaining energy, each having ~5.2% (~45 μJ per pulse). As such, the energy ratio of the strong and weak beams is approximately 4:1. The strong and weak beams have transmit energy levels to within approximately 10% of the mean values (i.e. 175 ± 17 μJ per pulse for the strong beams and 45 ± 5 μJ per pulse for the weak beams).

All other ICESat-2 ATL03-related information and specs can be found here:

https://nsidc.org/data/atl03

Click on the “Overview” or “User Guide” tabs for more info at the link above.

Thanks,
The PhoREAL Team

**Comment by zygym** [comment]
Hello！when I ues the PhoREAL tool to process ICESAT-2 data, the generated ATL08 CSV files hava a field named "heights", are they represent the h_te_mean or h_te_median or h_te_best_fit? Furthermore, some value of "heights" do not have specific value, how it happened and to how to solve them? Thank you very much!
![image](https://user-images.githubusercontent.com/62503050/91857679-c86e1e00-ec9a-11ea-9acd-6576c1c6e440.png)

**Comment by icesat-2UT** [comment]
Hi,

Sorry about that, this is a small error with PhoREAL. There are incorrect column labels in the ATL08 .csv file. This has been corrected internally and will be updated in a future release of PhoREAL.

For now, you will have to make the adjustments manually. In the ATL08 .csv file headers, Height = Max Canopy, Classification = Terrain Best Fit, and Signal Confidence = Terrain Median.

More details about ICESat-2 data products (ATL03, ATL08, etc.) can be found here: https://nsidc.org/data/icesat-2/data-sets

Again, sorry for the confusion in column headers.

Thanks,
PhoREAL Team

**Comment by zygym** [comment]
Copy that, thank you very much!

**Comment by zygym** [comment]
Hello! we have downloaded some ICESat-2 data from National Snow and Ice Data Center (ATL03_20181023195417_03850102_003_01.h5; ATL08_20181023195417_03850102_003_01.h5) we process them by PhoREAL and we met some matters. 
![image](https://user-images.githubusercontent.com/62503050/94336181-bbcbb400-0013-11eb-9165-5f99eb51f199.png)
![image](https://user-images.githubusercontent.com/62503050/94336222-fcc3c880-0013-11eb-9059-4e76036595ec.png)
as the pictures shows above, the ICESat-2 data(ATL08&ATL03) are over underestimated the true ground(terrain) height, we determine that was because of  the reference ellipsoid and the ITRF2014 frame.  We have defined the XY corrdinate is WGS84 and the Z coordiante is ITRF2014, but it still can not fit with the airbrone lidar data. as you can see the field value of ATL08 are lower than 0, under normal circumstances, we think that a height less than 0 means below sea level, but we don't know what a value less than 0 means and how to correct it in the ITRF2014 reference frame. If you can offer us some infromation about this we will be grateful! Thank you very much!
